File: src/main/java/org/digma/intellij/plugin/jaegerui/model/outgoing/SpanData.java
package org.digma.intellij.plugin.jaegerui.model.outgoing;


import com.fasterxml.jackson.annotation.*;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;

import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
@JsonSerialize
public record SpanData(@JsonProperty(value = "hasCodeLocation",required = true) boolean hasCodeLocation,
                       @JsonProperty(value = "insights",required = true) List<Insight> insights) {

    @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
    public SpanData {
        //nothing to do , for jackson only
    }
}

File: src/main/kotlin/org/digma/intellij/plugin/ui/jaegerui/model/SpanData.kt
package org.digma.intellij.plugin.ui.jaegerui.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import java.beans.ConstructorProperties


@JsonIgnoreProperties(ignoreUnknown = true)
data class SpanData
@JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
@ConstructorProperties("hasCodeLocation", "insights")
constructor(
    @get:JsonProperty("hasCodeLocation")
    @param:JsonProperty("hasCodeLocation")
    val hasCodeLocation: Boolean,
    val insights: MutableList<Insight>
)
